### PR TITLE
TOOL message type for Prompt

### PR DESCRIPTION
**Pull request by willbakst**
- Added a basic parser for the TOOL message type.
- Does not handle additional {id} or {name} fields that are available when using OpenAI.

### CODE REVIEW

File: mirascope/chat/tools.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 class OpenAITool(BaseModel):
     """A base class for more easily using tools with the OpenAI Chat client."""
 
-    model_config = ConfigDict(arbitrary_types_allowed=True)
+    model_config = ConfigDict(arbitrary_types_allowed=True, extra="allow")
```
Comment (brenkao) [inline mirascope/chat/tools.py:13]: why is this necessary.

**Review by brenkao — APPROVED** [review]

File: mirascope/chat/tools.py (hunk 1 above)
Comment (willbakst) [reply mirascope/chat/tools.py:13]: I did this so that the tool can have a `tool_call` attribute that we set after creating the tool specified by the consumer. See `chat.types.py` lines 64 and 84.

Another option would be to add a private `tool_call` attribute. The consumer of the defined tool would be able to pass in a tool call, which might actually be better. WDYT?

Comment (brenkao) [reply mirascope/chat/tools.py:13]: i would prefer the second option, having defined attributes and not allowing extras seems better practice

Comment (willbakst) [reply mirascope/chat/tools.py]: Done.

**Review by brenkao — APPROVED** [review]
